go/vt/discovery/healthcheck.go

```diff
@@ -625,14 +631,17 @@ func (hc *HealthCheckImpl) GetHealthyTabletStats(target *query.Target) []*Tablet
 	return append(result, hc.healthy[KeyFromTarget(target)]...)
 }
 
-// getTabletStats returns all tablets for the given target.
+// GetTabletStats returns all tablets for the given target.
 // The returned array is owned by the caller.
 // For TabletType_PRIMARY, this will only return at most one entry,
 // the most recent tablet of type primary.
-func (hc *HealthCheckImpl) getTabletStats(target *query.Target) []*TabletHealth {
+func (hc *HealthCheckImpl) GetTabletStats(target *query.Target) []*TabletHealth {
 	var result []*TabletHealth
 	hc.mu.Lock()
 	defer hc.mu.Unlock()
+	if target.Shard == "" {
+		target.Shard = "0"
+	}
```

Comment:
This looks like a bad thing to do. Why is the Shard not specified when we get the target?